numpy/lib/shape_base.py

```diff
@@ -405,18 +404,23 @@ def array_split(ary, indices_or_sections, axis=0):
         Ntotal = len(ary)
     try:
         # handle scalar case.
-        Nsections = len(indices_or_sections) + 1
-        div_points = [0] + list(indices_or_sections) + [Ntotal]
+        Nsections = len(chunks) + 1
+        div_points = [0] + list(chunks) + [Ntotal]
     except TypeError:
-        # indices_or_sections is a scalar, not an array.
-        Nsections = int(indices_or_sections)
-        if Nsections <= 0:
-            raise ValueError('number sections must be larger than 0.')
-        Neach_section, extras = divmod(Ntotal, Nsections)
-        section_sizes = ([0] +
-                         extras * [Neach_section+1] +
-                         (Nsections-extras) * [Neach_section])
-        div_points = _nx.array(section_sizes).cumsum()
+        # chunks is not iterable
+        try:
```

Comment:
The name 'indices_or_sections' is excessively verbose. In many similar situations other more established numpy functions use a more compact name. Additionally, there are many situations were it makes a lot of sense not to specify it.